networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider_nsx_policy.py

```diff
@@ -618,6 +617,33 @@ def _wait_to_realize(self, resource_type, os_id):
         exporter.REALIZED.labels(resource_type, status).inc()
         raise Exception("{} ID: {} did not get realized for {}s", resource_type, os_id, until * pause)
 
+    def _realize_bulk(self, resource_type: str, data, nsxt_res_type, dsl):
+        report = "Bulk Operation: Resource: {} with ID: {} is going to be %s.".format(resource_type, lambda: data[0]["id"])
+
+        LOG.info(report, "created")
+        path = "{}".format(API.INFRA)
+        
+        res = self.client.patch(path=path, data=data)
+        res.raise_for_status()
+
+        nsxt_objects = [child[nsxt_res_type] for child in data["children"][0]["children"]]
+        nsxt_objects = sorted(nsxt_objects, key=lambda x: x["id"])
+
+        # Fetch _revision data (needed to run put requests) for all updated objects
+        # This is needed to run put requests on objects since put relies on the correct _revision number
+        updated_nsxt_objs = self.client.get_all(path=API.SEARCH_DSL, params=API.SEARCH_DSL_QUERY(nsxt_res_type, dsl))
```

Comment:
I'm not sure about the search API approach. Ideally, we need to use GET /infra as we use the same endpoint for the update/create operation.